bvm/ballerina-core/src/main/java/org/ballerinalang/util/ResourceManager.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http:www.wso2.org) All Rights Reserved.
+ *  
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *  
+ *     http:www.apache.orglicensesLICENSE-2.0
+ *  
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.util;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * {@code {@link ResourceManager}} is a container for a {@code {@link Connection}}, a {@code {@link Statement}} and a
+ * {@code {@link Set}} of {@code {@link ResultSet}} associated with the {@code {@link Connection}}.
+ *
+ * @since 0.965.0
+ */
+public class ResourceManager {
+    private Connection connection;
+    private Statement statement;
+    private Set<ResultSet> resultSetSet;
+
+    public ResourceManager(Connection conn, Statement stmt) {
+        this.connection = conn;
+        this.statement = stmt;
+    }
+
+    /**
+     * Close the connection and the statement.
+     *
+     * @throws SQLException if an issue occur while closing the connection or statement
+     */
+    public void releaseResources() throws SQLException {
+        if (statement != null && !statement.isClosed()) {
+            statement.close();
+            statement = null;
+        }
+        if (connection != null && !connection.isClosed()) {
+            connection.close();
+            connection = null;
+        }
+    }
+
+    /**
+     * Add a {@code {@link ResultSet}} to the Set of result sets. This {@code {@link ResultSet}} should be a one
+     * associated with the connection. If an unrelated {@code {@link ResultSet}} is added it may result in undesirable
+     * consequences as when calling {@link #gracefullyReleaseResources(boolean)} releasing resources is based on
+     * whether all the result sets are closed.
+     *
+     * @param rs the result set to be added
+     */
+    public void addResultSet(ResultSet rs) {
+        if (resultSetSet == null) {
+            resultSetSet = new HashSet<>();
```

Comment:
Actually I did that on purpose to avoid a Set being initialized in scenarios where no result sets are added. I could have initialized it in the constructor with size 0 though (which would make it initialize with a capacity of 1)


+1 for the renaming. ResultSetSet is a Set of ResultSets, so may be its better to rename it as resultSets instead of resultSet.